packages/aws-client/lib/S3MultipartUploads.js

```diff
@@ -0,0 +1,115 @@
+// Utility functions to help with S3 multi-part uploads
+
+'use strict';
+
+const range = require('lodash/range');
+const { s3 } = require('../services');
+
+const GB = 1024 * 1024 * 1024;
+
+// Each part of a multi-part copy needs to specify a byte range to be copied.
+// This byte range has a starting byte and an ending byte (inclusive) that makes
+// up the part. All parts must be larger than 5 MB, except that the last part
+// may be smaller than 5 MB. Parts cannot be larger than 5 GB.
+//
+// This function takes a file size and an optional maxSize. It returns an array
+// of objects, each containing a `start` and an `end` value. These will make up
+// the ranges of the multi-part copy.
+//
+// https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html
+const createMultipartChunks = (size, maxSize = 5 * GB) => {
+  const lastChunkSize = size % maxSize;
```

Comment:
Don't we need to clamp `maxSize` (or `maxChunkSize` as suggested above) to the range [5 MB, 5 GB]? There's nothing here that would prevent specifying a size outside of this range.

Reply:
In an awesome twist, it turns out that you can use multi-part upload copies on files of any size. I was testing against the real AWS service and realized that I was able to upload a 10-byte file using multi-part. The documentation says that the only chunk that can be less than 5 MB is the last chunk. Turns out that, if you only send one chunk, it can be as small as you want.

I also did some performance testing against AWS and found that a chunk size of about 250 MB actually seems to give the best performance. Copying an 11 GB file using 5 GB chunks took about 133 seconds, while copying the same file with 250 MB chunks only took 10 seconds. Small sample size, but the smaller chunks did show an order of magnitude improvement.

To be on the safe side, I did update this function to throw an exception if someone tries to use a `maxSize` greater than 5GB.

Reply:
Ah, yes, that makes perfect sense, and nice speedup! See my other comment about either also throwing when chunk size is too small, or not throwing in either case.